src/containers/MyNdla/MyProfile/MyProfilePage.tsx

```diff
@@ -0,0 +1,164 @@
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { useContext, useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import styled from '@emotion/styled';
+import { spacing } from '@ndla/core';
+import { ButtonV2 } from '@ndla/button';
+import SafeLink from '@ndla/safelink';
+import { HelmetWithTracker, useTracker } from '@ndla/tracker';
+import {
+  ModalBody,
+  ModalCloseButton,
+  ModalHeader,
+  ModalTitle,
+  Modal,
+  ModalTrigger,
+  ModalContent,
+} from '@ndla/modal';
+import { Heading, Text } from '@ndla/typography';
+import InfoPart from '../InfoSection';
+import { AuthContext } from '../../../components/AuthenticationContext';
+import { constructNewPath, toHref } from '../../../util/urlHelper';
+import { useBaseName } from '../../../components/BaseNameContext';
+import { useDeletePersonalData } from '../userMutations';
+import { getAllDimensions } from '../../../util/trackingUtil';
+import MyPreferences from './components/MyPreferences';
+import MyContactArea from '../components/MyContactArea';
+import { UserInfo } from '../components/UserInfo';
+import MyNdlaPageWrapper from '../components/MyNdlaPageWrapper';
+
+const StyledPageContentContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  margin-top: ${spacing.normal};
+  gap: ${spacing.large};
+`;
+
+const ButtonRow = styled.div`
+  display: flex;
+  gap: ${spacing.small};
+  justify-content: flex-end;
+`;
+
+const InfoContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: ${spacing.normal};
+`;
+
+const ButtonContainer = styled.div`
+  display: flex;
+  align-items: baseline;
+  flex-direction: column;
+  gap: ${spacing.small};
+`;
+
+const MyProfilePage = () => {
+  const { user } = useContext(AuthContext);
+  const { t } = useTranslation();
+  const basename = useBaseName();
+  const location = useLocation();
+  const { trackPageView } = useTracker();
+  const { deletePersonalData } = useDeletePersonalData();
+
+  useEffect(() => {
+    trackPageView({
+      title: t('htmlTitles.myProfile'),
+      dimensions: getAllDimensions({ user }),
+    });
+  }, [t, trackPageView, user]);
+
+  const onDeleteAccount = async () => {
+    await deletePersonalData();
+    window.location.href = constructNewPath(
+      `/logout?state=${toHref(location)}`,
+      basename,
+    );
+  };
+
+  return (
+    <MyNdlaPageWrapper>
+      <StyledPageContentContainer>
+        <HelmetWithTracker title={t('myNdla.myProfile.title')} />
+        <Heading
+          element="h1"
+          id="myProfileTitle"
+          margin="none"
+          headingStyle="h1-resource"
+        >
+          {t('myNdla.myProfile.title')}
+        </Heading>
+        <MyContactArea user={user} />
+        <MyPreferences user={user} />
+        <InfoContainer>
+          {user && (
+            <InfoPart title={t('myNdla.myPage.feide')} icon={''}>
```

Comment:
Kan icon defineres slik? icon i InfoPart kunne vore definert med ? for å unngå dette.